### PR TITLE
Upgrade to kotlin 1.9

### PR DESCRIPTION
**Pull request by sstone**
This PR updates our code, builds scripts and publishing scripts to use kotlin 1.9.
It currently uses a snapshot version of `secp256k1-kmp` which will need to be updated once https://github.com/ACINQ/secp256k1-kmp/pull/92 has been merged and a new release published.

### CODE REVIEW

File: publishing/PUBLISHING.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Publishing bitcoin-kmp artifacts
+
+## snapshots
+
+Snapshots are published to the Sonatype snapshot repository (https://oss.sonatype.org/content/repositories/snapshots/).
+To publish snapshot, you must add your sonatype credentials for the `ossrh` server to your local maven settings (typically in $HOME/.m2/settings.xml)
```
Comment (t-bast) [inline publishing/PUBLISHING.md:6]: nit:
```suggestion
To publish a snapshot, you must add your sonatype credentials for the `ossrh` server to your local maven settings (typically in $HOME/.m2/settings.xml)
```

File: publishing/PUBLISHING.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+- Download `snapshot.zip` generated by the `Publish snapshot` github action
+- unzip `snapshot.zip` in the `publishing` directory
+- run `bitcoin-kmp-snapshot-deploy.sh`
+
+## releases
+
+Releases are published to the Sonatype staging repository. If all items are valid they will be published to `maven central` repository.
```
Comment (t-bast) [inline publishing/PUBLISHING.md:14]: nit:
```suggestion
Releases are published to the Sonatype staging repository. If all items are valid they will be published to the `maven central` repository.
```

File: src/commonMain/kotlin/fr/acinq/bitcoin/ScriptElt.kt [hunk 3]
```diff
@@ -558,11 +558,7 @@ public object ScriptEltMapping {
     @JvmField
     public val code2elt: Map<Int, ScriptElt> = elements.associateBy { it.code }
 
-    public fun name(elt: ScriptElt): String {
-        val name = elt.toString().removePrefix("fr.acinq.bitcoin.OP_")
-        val name1 = name.take(name.lastIndexOf('@'))
-        return name1
-    }
+    public fun name(elt: ScriptElt): String = elt.toString().removePrefix("OP_")
```
Comment (t-bast) [inline src/commonMain/kotlin/fr/acinq/bitcoin/ScriptElt.kt:561]: Nice, this feels cleaner :)

File: publishing/PUBLISHING.md (hunk 1 above)
Comment (sstone) [reply publishing/PUBLISHING.md]: done in https://github.com/ACINQ/bitcoin-kmp/pull/108/commits/e09c930887eb855e937cf7db3a8defb199943b70

File: publishing/PUBLISHING.md (hunk 2 above)
Comment (sstone) [reply publishing/PUBLISHING.md]: done in https://github.com/ACINQ/bitcoin-kmp/pull/108/commits/e09c930887eb855e937cf7db3a8defb199943b70

**Review by t-bast — APPROVED** [review]

**Pull request merged**